adapters/repos/db/lsmkv/segment.go

```diff
@@ -72,7 +72,7 @@ type diskIndex interface {
 
 func newSegment(path string, logger logrus.FieldLogger, metrics *Metrics,
```

Comment:
How is an overwriteDerived determined?

Reply:
overwriteDerived forces the generation of the bloom filter and other data that is generated based on the segment information.
Currently compaction pre-generates such information before the actual compacted segment is generated, in such a case, we don't want to overwrite the already calculated bloom filter. In other cases, we want to generate the bloom filter based on the new data.